src/chat.cpp

```diff
@@ -692,8 +704,14 @@ void CHAT_Render( void )
 
 	// Build the message that we will display to clients.
 	FString displayString = g_ChatBuffer.GetMessage();
+
+	// [AK] Split the message string into two components, from where the cursor is.
+	int cursorPosition = g_ChatBuffer.GetPosition();
+	FString messageFront = displayString.Mid( 0, cursorPosition );
+	FString messageBack = displayString.Mid( cursorPosition );
+
 	// Insert the cursor string into the message.
-	displayString = displayString.Mid( 0, g_ChatBuffer.GetPosition() ) + cursor + displayString.Mid( g_ChatBuffer.GetPosition() );
+	displayString = messageFront + cursor + messageBack;
```

Comment:
I may be overlooking something, it shouldn't it be sufficient simply to insert `cursor`  or `" "` depending on whether the cursor should be invisible or not? Then you shouldn't need to change anything else here.

Reply:
That is how I originally did it, but I found that the problem with doing it that way is that the widths of the graphics used for the '_' and " " characters of the SmallFont are different. So when the cursor is not at the end of the message, you'll notice that when the cursor switches between visible and invisible, everything behind the cursor would shift left and right, and this didn't look good.

Reply:
I see. There should be other invisible characters (e.g. tab). Do you know if all have them have same widths? If not, perhaps one of them has the same as `_`.

Reply:
I don't think there's anything for the tab, since it's a control code. I'm still not sure if it's a good idea to try using another invisible character as a substitute for the cursor when it's invisible. While they might have the same widths when using the standard SmallFont in Doom, what if somebody adds a replacement to the SmallFont and these character pairs end up having different widths?